docs/10-stories.md

```diff
@@ -0,0 +1,168 @@
+1. Each component should have a Demo story which is designed for documentation purposes and a series of stories detailing specific cases for visual regression testing
+1. Stories should have PascalCase names
+1. The `args` property should be passed to all stories so that the source code is displayed on the docs page
+1. Props should all be commented to autofill the args table
+1. Components should all have JSDoc comments to auto fill the docs page (see [example](#example) below)
+1. Controls should be disabled for all stories except the Demo story
+
+    ```ts
+    export const Story = ...
+
+    Story.parameters = {
+        controls: {
+            disabled: true
+        }
+    }
+    ```
+
+    or
+
+    ```ts
+    export const Story = ...
+
+    Story.story = {
+        parameters: {
+            controls: {
+                disabled: true
+            }
+        }
+    }
+    ```
+
+1. The Demo story should set all configurable args via the Story configuration and hide any other controls
+
+    ```ts
+    export const Story = ...
+
+    Story.story = {
+        args: {
+            arg1: "value"
+        },
+        argTypes: {
+            arg2: {
+                control: null
+            }
+        }
+    }
+    ```
+
+### Example
+
+```tsx
+// MyComponent.tsx
+
+export interface MyComponentProps {
+    /**
+     * A string to label the value
+     */
+    label: string;
+    /**
+     * An optional boolean to hide the label value
+     */
+    hideLabel?: boolean;
+    /**
+     * An optional string value to display. Will show `No value set` if not provided
+     */
+    value?: string;
+    /**
+     * An optional id to give the component
+     */
+    id?: string;
+}
+
+/**
+ * [Storybook](https://guardian.github.io/source/?path=/docs/source-${MY_COMPONENT_PACKAGE}-${MY_COMPONENT}--demo) •
+ * [Design System](https://theguardian.design/${MY_COMPONENT_PATH}) •
+ * [GitHub](https://github.com/guardian/source/tree/main/src/core/components/${MY_COMPONENT}) •
+ * [NPM](https://www.npmjs.com/package/@guardian/${MY_COMPONENT_PACKAGE})
+ *
+ * My Component is a nifty way of showing key -> value data
+ *
+ * The following themes are supported: `default`, `brand`
+ */
+export MyComponent = ({}: MyComponentProps) => (
+    <div id={id}>
+        {!hideLabel ? <div className="my-component-label">{label}</div> : null}
+        <div className="my-component-value">{value ?? "No value set"}</div>
+    </div>
+)
+
+
+
+
+// my-component.stories.tsx
+
+export default {
+	title: 'Source/src-my-component/MyComponent',
+	component: MyComponent,
+};
+
+export const Demo = (args: MyComponentProps) => (
+	<MyComponent {...args} />
+);
+
+Demo.story = {
+	argTypes: {
+		id: { control: null },
+	},
+	args: {
+		label: 'Label',
+		hideLabel: false,
+		value: 'Value',
+	},
+};
+
+export const NoValue = (args: MyComponentProps) => (
+	<MyComponent {...args} label="Label"/>
```

Comment:
If we're not passing default `args` via the `NoValue.args = {...}` pattern, is it useful to spread the args into the component like this? Is it doing anything?

Reply:
It's not, good point. The only thing that we need is to pass an arg into the function so that Storybook generates the source for the docs page. We could instead do:

```tsx
export const NoValue = (_: MyComponentProps) => (
  <MyComponent label="Label"/>
)
```

Reply:
Oh wow, that's unusual. So if you don't specifically define an argument, the docs don't get generated?

Reply:
Correct - for example:

## With Args

```tsx
export const LegendSupportingTextLight = (args: RadioGroupProps) => (
	<RadioGroup
		{...args}
		name="colours"
		label="Select your preferred colour"
		supporting="You can always change it later"
	>
		{radiosOneSelected.map((radio, index) =>
			React.cloneElement(radio, { key: index }),
		)}
	</RadioGroup>
);
```

or

```tsx
export const LegendSupportingTextLight = (_: RadioGroupProps) => (
	<RadioGroup
		name="colours"
		label="Select your preferred colour"
		supporting="You can always change it later"
	>
		{radiosOneSelected.map((radio, index) =>
			React.cloneElement(radio, { key: index }),
		)}
	</RadioGroup>
);
```

![image](https://user-images.githubusercontent.com/30179286/128021364-bd2997c2-70ea-40eb-8073-df8f9d6a29b2.png)

## Without Args

```tsx
export const LegendSupportingTextLight = () => (
	<RadioGroup
		name="colours"
		label="Select your preferred colour"
		supporting="You can always change it later"
	>
		{radiosOneSelected.map((radio, index) =>
			React.cloneElement(radio, { key: index }),
		)}
	</RadioGroup>
);
```

![image](https://user-images.githubusercontent.com/30179286/128021675-c3911029-82ce-4af2-ab34-1b0f1069476d.png)

Reply:
storybook must be able to introspect useful info from you explicitly giving the args type, even if you don't use it?

Reply:
Are we happy with this then?

```tsx
export const LegendSupportingTextLight = (_: RadioGroupProps) => (
	<RadioGroup
		name="colours"
		label="Select your preferred colour"
		supporting="You can always change it later"
	>
		{radiosOneSelected.map((radio, index) =>
			React.cloneElement(radio, { key: index }),
		)}
	</RadioGroup>
);
```